packages/design-system-tokens/src/lib/utility.ts

```diff
@@ -0,0 +1,87 @@
+import fs from 'fs';
+import path from 'path';
+import { HexValue, RGBValue, FileDescriptor } from './types';
+
+// converts an rgb string 'rgb(15,24,128)' to a hex value '#0819A9'
+export const rgbToHex = (r: number, g: number, b: number): HexValue => {
+  const hex = ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
+  return `#${hex}`;
+};
+
+// converts a hex string '#F3G1AA' to an rgb value string 'rgb(142, 24, 89)'
+export const hexToRgb = (hex: HexValue): RGBValue | null => {
+  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+  if (result) {
+    const r = parseInt(result[1], 16);
+    const g = parseInt(result[2], 16);
+    const b = parseInt(result[3], 16);
+    return `rgb(${r},${g},${b})`;
+  }
+  return null;
+};
+
+/*
+ * given a directory path string and an empty string array, returns a
+ * string array containing all files under the given path
+ */
+export const getAllFiles = (dirPath: string, arrayOfFiles: string[]): string[] => {
+  const files = fs.readdirSync(dirPath);
+
+  arrayOfFiles = arrayOfFiles || [];
+
+  files.forEach(function (file): void {
+    if (fs.statSync(`${dirPath}/${file}`).isDirectory()) {
+      arrayOfFiles = getAllFiles(`${dirPath}/${file}`, arrayOfFiles);
+    } else {
+      arrayOfFiles.push(path.join(dirPath, file));
+    }
+  });
+  return arrayOfFiles;
+};
+
+/*
+ * search out all availible modules under a path and return an
+ * array of objects which contains file descriptors for each file
+ */
+export const getFileDescriptors = (inPath: string): FileDescriptor[] => {
+  const root = `${process.cwd()}/src`;
+  const af = getAllFiles(`${root}/${inPath}`, []);
+  const fd: FileDescriptor[] = [];
```

Comment:
[nitpick] can these variable names be updated to be more descriptive?

Reply:
Sure, i'll make all these more descriptive

Reply:
I've updated all the opaquely named variables to be more descriptive!